### ISSUE TITLE
`ConnectionError(Timed out during query execution.)` during seeding

### ISSUE THREAD

**Issue by asciant** [prisma/prisma#21772]
### Bug description

When associating one record with another using `connect` the prisma client throws a connection error.

```
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: ConnectionError(Timed out during query execution.), transient: false })
```

### How to reproduce

Using the schema below.

```prisma
model Invoice {
  id String @id @default(cuid())

  amount     Int
  status     String
  date       String
  customer   Customer? @relation(fields: [customerId], references: [id])
  customerId String?
}

model Customer {
  id        String    @id @default(cuid())
  name      String
  email     String
  image_url String
  invoices  Invoice[]
}
```

Using this syntax results in a connection error:
```ts
prisma.invoice.create({
    data: {
        customer: {
            connect: {
                id:
            },
        },
        amount: invoice.amount,
        status: invoice.status,
        date: invoice.date,
    },
});
```

Does not result in a connection error:
```ts
prisma.invoice.create({
    data: {
        customerId: invoice.customer_id,
        amount: invoice.amount,
        status: invoice.status,
        date: invoice.date,
    },
});
```

### Expected behavior

It should make the connection using the preferred syntax, as per the `nested connect query API`.

### Prisma information

[Gist](https://gist.github.com/asciant/5f5bf1753bfb493a5b45d3fd0e43a93a)

### Environment & setup
```
- OS: macOS
- Database: Turso
- Node.js version: v21.1.0
```

### Prisma Version

```
prisma                  : 5.5.2
@prisma/client          : 5.5.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/.pnpm/@prisma+engines@5.5.2/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/.pnpm/@prisma+engines@5.5.2/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.5.1-1.aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Default Engines Hash    : aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Studio                  : 0.494.0
Preview Features        : driverAdapters
```

**Comment by janpio** [comment]
Is the problem in your seed.ts script?
If so, that is not actually using the driver adapter as it just does `new PrismaClient();` without supplying an adpater instance that has the LibSQL/Turso driver.

**Comment by asciant** [comment]
That was it thanks Jan, I updated the `seed.ts` to use the adapter instance and it seeded as expected. Not sure how the connection error fits in, if the seed script was only connecting to my local sqlite database `DATABASE_URL="file:./dev.db"`.

I can close this issue if you're happy to?

**Comment by janpio** [comment]
We should probably reproduce this in SQLite context to see if we get the same problem.

**Comment by asciant** [comment]
It may well be that using the `driverAdapaters` preview feature flag (`previewFeatures = ["driverAdapters"]`) and then generating the Prisma client was a factor. Given that the `DATABASE_URL` environment variable was `DATABASE_URL="file:./dev.db"` (a local DB).

**Comment by miguelff** [comment]
The issue is not reproducible with local SQLite, using the traditional prisma client (no driver adapters) against a local db.

**Comment by Zn4rK** [comment]
I ran into this error today, with a local SQLite, using the traditional prisma client (no driver adapters) against a local db.

The only difference in reproduction would be to loop the connect a few times (to simulate multiple connections - which is how I encountered it - multiple writes concurrently):

```javascript
for (let i = 0; i < 10; i++) {
  prisma.invoice.create({
    data: {
        customer: {
            connect: {
                id,
            },
        },
        amount: invoice.amount,
        status: invoice.status,
        date: invoice.date,
    },
  });
}
```

**Comment by Druue** [comment]
I can't seem to reproduce this @Zn4rK, you can find my repro attempt [here](https://github.com/Druue/prisma-debug/tree/repro/21772)

**Comment by jkomyno** [comment]
Hi @asciant and @Zn4rK, we believe we have a fix for this issue.

We plan on releasing it with the next stable version of Prisma (`5.16.0`), but in the meantime, we'd like to collect some early feedback directly from the impacted users. Can you please try installing Prisma `5.16.0-integration-sqlite-fix.1`, and confirm that you can no longer experience this bug with it? Thanks!

```sh
pnpm install -D prisma@5.16.0-integration-sqlite-fix.1
pnpm install @prisma/client@5.16.0-integration-sqlite-fix.1
```

(Please note that this Prisma version is __unstable__, so we don't recommend running it in production.)

**Comment by jkomyno** [comment]
Hello! This issue was solved via https://github.com/prisma/prisma-engines/pull/4907. The fix is already available on Prisma 5.15.1.

Please make sure you update the `prisma` CLI, the `@prisma/client` library, and run `prisma generate` again :)
